tests/rptest/tests/cluster_quota_test.py

```diff
@@ -100,3 +102,114 @@ def test_partition_throttle_mechanism(self):
         assert foo_response[0]['ErrorCode'] == 0  # success
         assert baz_response[0]['ErrorCode'] == 89  # throttling_quota_exceeded
         assert response['ThrottleMillis'] > 0
+
+
+class ClusterRateQuotaTest(RedpandaTest):
+    """
+    Ducktape tests for rate quota
+    """
+    topics = (TopicSpec(), )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args,
+                         num_brokers=3,
+                         resource_settings=ResourceSettings(num_cpus=1),
+                         **kwargs)
+
+    def produce(self, producer, amount):
+        response_futures = [
+            producer.send(self.topic, self.msg) for _ in range(amount)
+        ]
+        for f in response_futures:
+            f.get(1)
+
+    @cluster(num_nodes=3)
+    def test_client_group_rate_throttle_mechanism(self):
+        """
+        Ensure group rate throttle works
+        """
+
+        self.redpanda.set_cluster_config({
+            "kafka_client_group_byte_rate_quota": [
+                {
+                    "group_name": "group_1",
+                    "clients_prefix": "producer_group_alone_producer",
+                    "quota": 10240
+                },
+                {
+                    "group_name": "group_2",
+                    "clients_prefix": "producer_group_multiple",
+                    "quota": 10240
+                },
+            ],
+            "max_kafka_throttle_delay_ms":
+            10,
+        })
+
+        producer = KafkaProducer(
+            acks="all",
+            bootstrap_servers=self.redpanda.brokers_list(),
+            value_serializer=str.encode,
+            retries=1,
+            client_id="producer_group_alone_producer")
+
+        self.msg = "".join(
+            random.choice(string.ascii_lowercase) for _ in range(1024))
+
+        # Produce under the limit
+        self.produce(producer, 10)
+        assert producer.metrics(
+        )["producer-metrics"]["produce-throttle-time-max"] == 0
+
+        # Produce more than limit
+        self.produce(producer, 100)
+        assert producer.metrics(
+        )["producer-metrics"]["produce-throttle-time-max"] > 0
```

Comment:
maybe > 0 && < 10? to make sure the calculation applied boundry is applied correctly.

Reply:
fixed